File: 03-CPP/06-arreglos/01arrays.cpp
// Arreglos en C++
#include <iostream>
using namespace std;

int main()
{
    // Declaración de arrays
    int numerosArray[5];
    numerosArray[0] = 13;
    numerosArray[1] = 21;
    numerosArray[4] = 62;

    cout << "Valor 1: " << numerosArray[0] << endl;
    cout << "Valor 2: " << numerosArray[1] << endl;
    cout << "Valor 3: " << numerosArray[3] << endl;

    // Declaracion simplificada de arrays.
    int numeros[] = {2, 3, 5, 7};
    cout << "Valor 1 de numeros: " << numeros[0] << endl;
    cout << "Valor 2 de numeros: " << numeros[1] << endl;
    cout << "Valor 5 de numeros: " << numeros[4] << endl; // Valor basura, no es valido
    cout << endl;

    // Iteracion de arrays
    
    // Calcular el largo de un array
    int largoArray = sizeof(numeros) / sizeof(numeros[0]);
    cout << "Largo del array: " << largoArray << endl;

    cout << "Iteracion de arrays" << endl;
    for (int i = 0; i < largoArray; i++)
    {
        cout << "Valor " << i << ": " << numeros[i] << endl;
    }

  




    return 0;
}

File: 04-CSHARP/06-Arreglos/01arrays.cs
// Arrays en C#

// Declaración de arrays 
int[] numerosArray;
numerosArray = new int[5];

numerosArray[0] = 13;
numerosArray[1] = 21;
numerosArray[4] = 62;

Console.WriteLine($"Valor 1: {numerosArray[0]}");
Console.WriteLine($"Valor 2: {numerosArray[1]}");
Console.WriteLine($"Valor 3: {numerosArray[2]}");
Console.WriteLine($"Valor 5: {numerosArray[4]}");

// Declaracion simplificada de arrays 
int[] numeros = { 100, 200, 300, 400 };
Console.WriteLine($"Valor 1: {numeros[0]}");
Console.WriteLine($"Valor 2: {numeros[1]}");
Console.WriteLine($"Valor 3: {numeros[2]}");
// Console.WriteLine($"Valor 4: {numeros[4]}");


// Iteracion de arrays 

// Largo del array 
Console.WriteLine($"Largo del array {numeros.Length}");

Console.WriteLine("Iteracion del array");
for (int i = 0; i < numeros.Length; i++)
{
    Console.WriteLine($"Valor del array: {numeros[i]}");
}

// ForEach 
Console.WriteLine("For mejorado");
foreach (var numero in numeros)
{
    Console.WriteLine($"Valor: {numero}");
}
